fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/IteratorCursor.java

```diff
@@ -37,14 +39,40 @@
  */
 @API(API.Status.MAINTAINED)
 public class IteratorCursor<T> extends IteratorCursorBase<T, Iterator<T>> {
+    private static final Logger LOG = LoggerFactory.getLogger(IteratorCursor.class);
+    private Closeable closeable;
+
     public IteratorCursor(@Nonnull Executor executor, @Nonnull Iterator<T> iterator) {
         super(executor, iterator);
     }
 
+    /**
+     * Adds an additional closeable that fires prior to the super.close().  Errors are logged but <b>are</b> swallowed.
+     *
+     * @param executor executor
+     * @param iterator iterator
+     * @param closeable closeable
+     */
+    public IteratorCursor(@Nonnull Executor executor, @Nonnull Iterator<T> iterator, Closeable closeable) {
+        super(executor, iterator);
+        this.closeable = closeable;
+    }
+
     @Nonnull
     @Override
     public CompletableFuture<RecordCursorResult<T>> onNext() {
         return CompletableFuture.completedFuture(computeNextResult(iterator.hasNext()));
     }
 
+    @Override
+    public void close() {
+        try {
+            if (closeable != null) {
+                closeable.close();
+            }
+        } catch (Exception ioe) {
+            LOG.warn("close failed",ioe);
```

Comment:
This should use a `KeyValueLogMessage` to match our other log messages, though it may not make much of a difference if it's just the static message anyway